app/assets/javascripts/details/state/edit_mode.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+var EditStateController = function (el) {
```

Comment:
@nickbristow This is where it gets used: https://github.com/18F/C2/blob/new-details-page/edit-mode/app/assets/javascripts/details/app_details.js